<a href="https://colab.research.google.com/github/shivaniii24/Parkinson-s-Disease-Classification/blob/main/Parkinson%E2%80%99s_Disease_Classification.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***Parkinson’s Disease Classification***

Parkinson's is a disease that can cause a
nervous system disorder and affects the movement. Parkinson dataset contains biomedical measurements, 195 records of people with 23 different attributes. This data is used to differentiate healthy people and people with Parkinson’s disease.

### Build a model that can be used to differentiate healthy people from people having Parkinson’s disease.

Import our libraries

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, accuracy_score

Read in our dataset

In [ ]:
df= pd.read_csv("/content/pd_speech_features.csv")

In [ ]:
df

id  gender  ...  tqwt_kurtosisValue_dec_36  class
0      0       1  ...                    18.9405      1
1      0       1  ...                    45.1780      1
2      0       1  ...                     4.7666      1
3      1       0  ...                     4.0603      1
4      1       0  ...                     6.1164      1
..   ...     ...  ...                        ...    ...
751  250       0  ...                     6.8671      0
752  250       0  ...                    28.4170      0
753  251       0  ...                     5.0424      0
754  251       0  ...                     3.7131      0
755  251       0  ...                     3.1527      0

[756 rows x 755 columns]

Since our data has some null values and many are close to zero.So let's replace all the null values to 0.

In [ ]:
df.replace(np.nan,0)

id  gender  ...  tqwt_kurtosisValue_dec_36  class
0      0       1  ...                    18.9405      1
1      0       1  ...                    45.1780      1
2      0       1  ...                     4.7666      1
3      1       0  ...                     4.0603      1
4      1       0  ...                     6.1164      1
..   ...     ...  ...                        ...    ...
751  250       0  ...                     6.8671      0
752  250       0  ...                    28.4170      0
753  251       0  ...                     5.0424      0
754  251       0  ...                     3.7131      0
755  251       0  ...                     3.1527      0

[756 rows x 755 columns]

Let's check if there is any null value left.

In [ ]:
df.isnull().sum()

id                           0
gender                       0
PPE                          0
DFA                          0
RPDE                         0
                            ..
tqwt_kurtosisValue_dec_33    0
tqwt_kurtosisValue_dec_34    0
tqwt_kurtosisValue_dec_35    0
tqwt_kurtosisValue_dec_36    0
class                        0
Length: 755, dtype: int64

Now our data is preparted to be seperated in x and y that is input and out.

In [ ]:
x=df.iloc[:,:754]
y=df['class'].values

In [ ]:
x

id  gender  ...  tqwt_kurtosisValue_dec_35  tqwt_kurtosisValue_dec_36
0      0       1  ...                     3.0004                    18.9405
1      0       1  ...                     6.3431                    45.1780
2      0       1  ...                     3.1495                     4.7666
3      1       0  ...                     6.2650                     4.0603
4      1       0  ...                     5.0559                     6.1164
..   ...     ...  ...                        ...                        ...
751  250       0  ...                     3.2003                     6.8671
752  250       0  ...                    10.9058                    28.4170
753  251       0  ...                     3.3545                     5.0424
754  251       0  ...                     2.8332                     3.7131
755  251       0  ...                     2.6217                     3.1527

[756 rows x 754 columns]

In [ ]:
y

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0,
       0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1,
       1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
       0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

Now we need to train our model and then test it.

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

In [ ]:
sc = StandardScaler()

In [ ]:
x_train = sc.fit_transform(x_train)
x_test = sc.fit_transform(x_test)

In [ ]:
classifier = LogisticRegression()

Fit the model

In [ ]:
classifier.fit(x_train, y_train)

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

Now that you have a training and testing dataset,its time to predict the values of our model

In [ ]:
pred_y = classifier.predict(x_test)

In [ ]:
pred_y

array([0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0,
       1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1,
       1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0,
       1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1,
       1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1])

The confusion matrix is a summary of prediction results for a given classification problem. The confusion matrix shows the number of correct and incorrect predictions broken down by each class.

In [ ]:
confusion_matrix(y_test,pred_y)

array([[ 26,  12],
       [ 14, 100]])

Here everything is done so lets check the accuracy of our model.

In [ ]:
accuracy_score = accuracy_score(y_test, pred_y)

In [ ]:
print("accuracy = ", accuracy_score * 100, "%")

accuracy =  82.89473684210526 %


## **Conclusion**
Here i have used logistic Regression and it gave me 82.89 almost 83% of accuracy.so my model can predict differentiate healthy people from people having Parkinson’s disease.